app/graphql/types/system.rb

```diff
@@ -67,8 +67,7 @@ def rule_objects_failed
     def last_scanned(args = {})
       if args[:profile_id].present?
         rule_ids = ::Profile.find(args[:profile_id]).rules.pluck(:id)
-        rule_results = ::RuleResult.where(rule_id: rule_ids,
-                                          host: object.id)
+        rule_results = object.rule_results.where(rule_id: rule_ids)
```

Comment:
I thought this would convert the query into a left outer join, but AR is doing it's magic well here and translating the query to the original one here. Why changing it if it's the same? Is it not clear enough?

```ruby
irb(main):005:0> Host.last.rule_results.where(rule_id: ["4d00711c-e8c7-430a-9940-67aa79c558e6"]).to_sql
=> "SELECT \"rule_results\".* FROM \"rule_results\" WHERE \"rule_results\".\"host_id\" = 'fe60873d-6585-4226-b7d3-b3d43b020f8e' AND \"rule_results\".\"rule_id\" = '4d00711c-e8c7-430a-9940-67aa79c558e6'" 

irb(main):015:0> rule_results = ::RuleResult.where(rule_id: '4d00711c-e8c7-430a-9940-67aa79c558e6', host_id: Host.last.id).to_sql
=> "SELECT \"rule_results\".* FROM \"rule_results\" WHERE \"rule_results\".\"rule_id\" = '4d00711c-e8c7-430a-9940-67aa79c558e6' AND \"rule_results\".\"host_id\" = 'fe60873d-6585-4226-b7d3-b3d43b020f8e'"


```

Reply:
Yeah, just a refactor to make it a bit cleaner. I can revert it if you have strong feelings.

Reply:
No problem, no need to change it if it's easier to read. I probably created this in the first place because I was not aware of the AR magic optimization.